### PR TITLE
PRC-108: Add references code API

### PR DESCRIPTION
**Pull request by james-baker-moj**
Simple API to get by group code.

Lift and shift from BVLS with a few extra tests

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ReferenceCodeService.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.mapping.toModel
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.model.response.ReferenceCode
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.repository.ReferenceCodeRepository
+
+@Service
+class ReferenceCodeService(private val referenceCodeRepository: ReferenceCodeRepository) {
+  fun getReferenceDataByGroup(groupCode: String): List<ReferenceCode> =
+    referenceCodeRepository.findAllByGroupCodeEquals(groupCode).toModel()
```
Comment (prabash-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ReferenceCodeService.kt:11]: would be good to cache it? [@Cacheable(](https://github.com/ministryofjustice/prison-api/blob/24ff92a56eac4f02e3101aa6cb1e0ed3a9ca46be/src/main/java/uk/gov/justice/hmpps/prison/repository/ReferenceDataRepository.java#L42)

**Review by prabash-moj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/entity/ReferenceCodeEntity.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+) {
+  var amendedBy: String? = null
+    private set
+
+  var amendedTime: LocalDateTime? = null
+    private set
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
```
[the remark this answers is not in the document]
Comment (james-baker-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/entity/ReferenceCodeEntity.kt:36]: No I just noticed that myself and was stripping it to make it a data class.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ReferenceCodeService.kt (hunk 1 above)
Comment (james-baker-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ReferenceCodeService.kt:11]: yeah - need to think about caching, maybe do in the front end instead?

Comment (prabash-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/service/ReferenceCodeService.kt:11]: good point, for this, may be do it on the frontend as it is just static data and not heavy computations involved here, But if we need to reduce db calls and improve performance, scalability, increase security etc then best to go with hybrid approach?

**Pull request merged**